### PR TITLE
feat(ci): Shell formatting

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The repo currently lacks a shell formatter.  As sis typical when a formatter is missing this is causing spurious diffs.

# Changes
- Add the shell formatter used in the nns-dapp (and other GIX projects).

# Tests
- CI already runs the top-level format check.

### CODE REVIEW

File: scripts/format.sh.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (bitdivine) [inline scripts/format.sh.sh:1]: This is copied verbatim from nns-dapp.

File: .github/workflows/backend-checks.yml [hunk 2]
```diff
@@ -29,6 +29,8 @@ jobs:
             ~/.cargo/git
             target
```
Comment (bitdivine) [inline .github/workflows/backend-checks.yml:30]: Most changes are just formatting.  The only other changes are this here and others highlighted with comments below.

File: scripts/format.sh [hunk 3]
```diff
@@ -6,4 +6,5 @@ set -x
 time xargs -P8 -I{} bash -c "{}" <<EOF
```
Comment (bitdivine) [inline scripts/format.sh:6]: Here we format shell scripts by default.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

Should we add a new GitHub Actions checks.yml that runs when a modification is applied in docker or scripts? `backend-checks.yml` runs only when a changes is applied to backend related things while most scripts are not necesserely related with the backend, more local development or build scripts. WDYT?

**Comment by bitdivine** [comment]
@peterpeterparker Thanks for the review.  I agree.  A formatting checks are fast so I would even potentially just run the check every time.  KISS :-)

**Pull request merged**